src/panel/context/Devtools.tsx

```diff
@@ -66,22 +76,51 @@ export const DevtoolsProvider: FC = ({ children }) => {
 
   // Listen for client init connection
   useEffect(() => {
-    addMessageHandler((message) => {
+    return addMessageHandler(async (message) => {
       if (message.type === "init") {
         setClientConnected(true);
-      } else if (message.type === "disconnect") {
+      }
+
+      if (message.type === "disconnect") {
         setClientConnected(false);
       }
     });
   }, [addMessageHandler, setClientConnected]);
 
+  // Check version on client connected
+  useEffect(() => {
+    if (!clientConnected) {
+      setVersion(({ required }) => ({
+        required,
+        mismatch: false,
+      }));
+    }
+
+    getExchangeVersion().then((v) => {
+      setVersion(({ required }) => ({
+        required,
+        mismatch:
+          !semver.valid(v) || !semver.satisfies(v as string, `>=${required}`),
+        actual: v,
+      }));
+    });
+  }, [clientConnected]);
+
   return (
     <DevtoolsContext.Provider
-      value={{ sendMessage, addMessageHandler, clientConnected }}
+      value={{ sendMessage, addMessageHandler, clientConnected, version }}
     >
       {children}
     </DevtoolsContext.Provider>
   );
 };
 
 let index = 0;
+
+const getExchangeVersion = () =>
+  new Promise<string | undefined>((resolve) =>
+    chrome.devtools.inspectedWindow.eval(
+      "window.__urql__",
```

Comment:
I'm going to move this to be devtools specific (i.e. something like `window.__urql_devtools__`)